internal/integrations/integrations.go

```diff
@@ -0,0 +1,142 @@
+	// until Component.Shutdown() ends.
+	GetExtensions() map[component.ID]component.Component
+}
+
+// Compile-time check to ensure that ExtensionGetter is a subset of the component.Host interface.
+var _ ExtensionGetter = (component.Host)(nil)
+
+// Find looks for integrations in extensions of the host that implement the IntegrationFinder interface.
+// It returns ErrNotFound if the integration cannot be found, or ErrNotConfigured if there are no
+// configured finder extensions. ErrNotConfigured also wraps ErrNotFound.
+func Find(ctx context.Context, logger *zap.Logger, host ExtensionGetter, integrationName string) (Integration, error) {
+	if host == nil {
+		logger.Error("received nil host")
+		return nil, ErrNotConfigured
+	}
+	anyExtension := false
+	for eid, extension := range host.GetExtensions() {
+		finder, ok := extension.(Finder)
+		if !ok {
+			continue
+		}
+		anyExtension = true
+
+		integration, err := finder.FindIntegration(ctx, integrationName)
+		if errors.Is(ErrNotFound, err) {
+			continue
+		}
+		if err != nil {
+			logger.Error("integration finder failed",
+				zap.String("component", eid.String()),
+				zap.Error(err))
+			return nil, err
+		}
+
+		return integration, nil
+	}
+	if !anyExtension {
+		return nil, ErrNotConfigured
+	}
+
+	return nil, ErrNotFound
+}
+
+// Config contains an structured integration.
+type Config struct {
+	Receivers  map[component.ID]ComponentConfig `mapstructure:"receivers"`
+	Processors map[component.ID]ComponentConfig `mapstructure:"processors"`
+	Pipelines  map[pipeline.ID]PipelineConfig   `mapstructure:"pipelines"`
+}
+
+// validate validates that the integration configuration is valid and all the components referenced in the
+// pipelines are defined.
+func (c *Config) validate() error {
+	if len(c.Pipelines) == 0 {
+		return errors.New("missing pipelines")
+	}
+	for _, pipeline := range c.Pipelines {
+		if pipeline.Receiver != nil {
+			if _, found := c.Receivers[*pipeline.Receiver]; !found {
+				return fmt.Errorf("receiver %s not defined", pipeline.Receiver.String())
+			}
+		}
+
+		for _, processor := range pipeline.Processors {
+			if _, found := c.Processors[processor]; !found {
+				return fmt.Errorf("processor %q not defined", processor.String())
+			}
+
+		}
+	}
+
+	return nil
+}
+
+// PipelineConfig contains the definition of a pipeline in the integration.
+type PipelineConfig struct {
+	// Receiver is the receiver to be used in the pipeline. It is optional, if a pipeline does not define a
+	// receiver it cannot be used to instantiate receivers.
+	Receiver *component.ID `mapstructure:"receiver"`
+
+	// Processors is the chain of processors of the pipeline, to be used as part of the receiver in receiver
+	// components, or as a combined processor when the pipeline is used as processor.
+	Processors []component.ID `mapstructure:"processors"`
+}
+
+// ComponentConfig contains the configuration of components. It is mainly irrelevant at this point.
```

Comment:
This comment needs adjustment.

If it is irrelevant, then it should not be used in other places like this:

```
type Config struct {
	Receivers  map[component.ID]ComponentConfig `mapstructure:"receivers"`
	Processors map[component.ID]ComponentConfig `mapstructure:"processors"`
	Pipelines  map[pipeline.ID]PipelineConfig   `mapstructure:"pipelines"`
}
```